src/fp2/dusk.rs

```diff
@@ -6,105 +6,83 @@
-                    where
-                        E: serde::de::Error,
-                    {
-                        match value {
-                            "c0" => Ok(Field::C0),
-                            "c1" => Ok(Field::C1),
-                            _ => Err(serde::de::Error::unknown_field(value, FIELDS)),
+                fn visit_map<A: MapAccess<'de>>(self, mut map: A) -> Result<Self::Value, A::Error> {
+                    let (mut c0, mut c1) = (None, None);
+                    while let Some(key) = map.next_key()? {
+                        match key {
+                            "c0" => {
+                                if c0.is_some() {
+                                    return Err(SerdeError::duplicate_field("c0"));
+                                } else {
+                                    c0 = Some(map.next_value()?);
+                                }
+                            }
+                            "c1" => {
+                                if c1.is_some() {
+                                    return Err(SerdeError::duplicate_field("c1"));
+                                } else {
+                                    c1 = Some(map.next_value()?);
+                                }
+                            }
+                            field => return Err(SerdeError::unknown_field(field, &FIELDS)),
                         }
                     }
+                    Ok(Fp2 {
+                        c0: c0.ok_or_else(|| SerdeError::missing_field("c0"))?,
+                        c1: c1.ok_or_else(|| SerdeError::missing_field("c1"))?,
+                    })
                 }
-
-                deserializer.deserialize_identifier(FieldVisitor)
-            }
-        }
-
-        struct Fp2Visitor;
-
-        impl<'de> Visitor<'de> for Fp2Visitor {
-            type Value = Fp2;
-
-            fn expecting(&self, formatter: &mut ::core::fmt::Formatter) -> ::core::fmt::Result {
-                formatter.write_str("struct Fp2")
             }
 
-            fn visit_seq<V>(self, mut seq: V) -> Result<Fp2, V::Error>
-            where
-                V: serde::de::SeqAccess<'de>,
-            {
-                let c0 = seq
-                    .next_element()?
-                    .ok_or_else(|| serde::de::Error::invalid_length(0, &self))?;
-                let c1 = seq
-                    .next_element()?
-                    .ok_or_else(|| serde::de::Error::invalid_length(0, &self))?;
-                Ok(Fp2 { c0, c1 })
-            }
+            const FIELDS: &[&str] = &["c0", "c1"];
+            deserializer.deserialize_struct("Fp2", FIELDS, Fp2Visitor)
         }
-
-        const FIELDS: &[&str] = &["c0", "c1"];
-        deserializer.deserialize_struct("Fp2", FIELDS, Fp2Visitor)
     }
-}
 
-#[test]
-fn fp2_serde_roundtrip() {
-    use crate::fp::Fp;
-    use bincode;
+    #[test]
+    fn fp2_serde_roundtrip() {
```

Comment:
```suggestion
    fn serde_fp2() {
```

Let's also align this test case name with the names of the other equivalent tests.